### PR TITLE
Catching LearningTask events.

### PR DESCRIPTION
**Pull request by lukaDoric**

### CODE REVIEW

File: src/app/modules/learning/task/task.component.html [hunk 1]
```diff
@@ -26,7 +26,7 @@ <h3>{{ task.name }}</h3>
             </div>
         </div>
         <mat-divider vertical="true"></mat-divider>
-        <mat-tab-group *ngIf="selectedStep" style="width: 75%;" dynamicHeight animationDuration="0"
+        <mat-tab-group #tabGroup (focusChange)="onTabChanged($event)" *ngIf="selectedStep" style="width: 75%;" dynamicHeight animationDuration="0"
```
Comment (Luburic) [inline src/app/modules/learning/task/task.component.html:29]: Do we need the `#tabGroup`?

File: src/app/modules/learning/task/task.component.ts [hunk 2]
```diff
@@ -44,6 +45,32 @@ export class TaskComponent implements OnInit {
...
 
+  public onTabChanged(tabChangeEvent: MatTabChangeEvent): void {
+    if(tabChangeEvent.index === 0) {
+      this.progressService.submissionOpened(this.task.unitId, this.task.id, this.taskProgress.id, this.selectedStep.id)
+      .subscribe();
+    } else if(tabChangeEvent.index === 1) {
+      this.progressService.guidanceOpened(this.task.unitId, this.task.id, this.taskProgress.id, this.selectedStep.id)
+      .subscribe();
+    } else if(tabChangeEvent.index === 2) {
+      this.progressService.exampleOpened(this.task.unitId, this.task.id, this.taskProgress.id, this.selectedStep.id)
```
Comment (Luburic) [inline src/app/modules/learning/task/task.component.ts:56]: Don't forget that there can be multiple videos at this level. In this file, look at the last function `getNextExample`.

**Review by Luburic — CHANGES_REQUESTED** [review]
Two additional cases for video-related events to address. Do not forget the `exampleOpened` event.

File: src/app/modules/learning/task/task.component.ts (hunk 2 above)
Comment (lukaDoric) [reply src/app/modules/learning/task/task.component.ts:56]: onVideoStatusChanged function captures events for any video.

**Review by Luburic — APPROVED** [review]
Not a fan of using query parameters to send the video URL when we have a null POST body, but we can merge.

**Pull request merged**